pkg/reconciler/pipelinerun/resources/pipelinerunstate.go

```diff
@@ -198,10 +198,18 @@ func (state PipelineRunState) GetRunsResults() map[string][]v1beta1.CustomRunRes
 
 // GetChildReferences returns a slice of references, including version, kind, name, and pipeline task name, for all
 // TaskRuns and Runs in the state.
-func (state PipelineRunState) GetChildReferences() []v1beta1.ChildStatusReference {
+func (facts *PipelineRunFacts) GetChildReferences() []v1beta1.ChildStatusReference {
 	var childRefs []v1beta1.ChildStatusReference
 
-	for _, rpt := range state {
+	for _, rpt := range facts.State {
+		// try to replace the parameters of the reference result of whenexpression in the taskrun that has ended
+		if rpt.TaskRun != nil && rpt.isDone(facts) {
+			resolvedResultRefs, _, err := ResolveResultRefs(facts.State, PipelineRunState{rpt})
+			if err == nil {
+				ApplyTaskResults(facts.State, resolvedResultRefs)
```

Comment:
I'm not sure if applying task results in `GetChildReferences` is a good approach, this looks like every time we get ChildReferences we need to apply the results. Is it possible to only apply at one place? (Sorry if I'm wrong)

Reply:
It should be applying task results in GetChildReferences only when taskRun isDone, so it may not  need apply the results every time